src/main/java/com/yahoo/bullet/rest/query/PubSubReader.java

```diff
@@ -80,4 +84,16 @@ public void run() {
         }
         subscriber.close();
     }
+
+    private static boolean isDone(PubSubMessage response) {
+        boolean done = false;
```

Comment:
This method can be defined as
``` return response.hasSignal() && FINISHED.contains(response.getMetadata().getSignal())```
where FINISHED is a static final Set with COMPLETE, KILL and FAIL

Reply:
Done.